### ISSUE TITLE
systemd-networkd won't add static routes if ppp interface is added after systemd-networkd has started

### ISSUE THREAD

**Issue by poscat0x04** [systemd/systemd#26356]
### systemd version the issue has been seen with

251.10

### Used distribution

NixOS 22.11

### Linux kernel version used

6.1.10

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

I'm using systemd-networkd to configure a ppp interface added by pppd. This is the `.network` file for it:

```
[Match]
Type=ppp

[Network]
DHCP=ipv6
IPv6AcceptRA=true
KeepConfiguration=static
DNS=127.0.0.1:53

[Route]
Destination=0.0.0.0/0
Metric=5

[Route]
Destination=0.0.0.0/0
Table=symmetry

[Route]
Destination=47.74.0.0/18
Table=others-direct

[Route]
Destination=47.74.32.0/19
Table=others-direct

[Route]
Destination=47.91.0.0/19
Table=others-direct

[Route]
Destination=47.91.16.0/20
Table=others-direct

[Route]
Destination=47.245.0.0/18
Table=others-direct

[Route]
Destination=47.245.32.0/19
Table=others-direct

[Route]
Destination=47.245.48.0/20
Table=others-direct

[Route]
Destination=8.209.192.0/18
Table=others-direct

[Route]
Destination=8.209.224.0/19
Table=others-direct

[Route]
Destination=8.211.128.0/18
Table=others-direct

[Route]
Destination=101.101.101.101
Table=others-direct

[Route]
Destination=101.6.6.6
Table=others-direct

[Route]
Destination=190.115.31.142
Table=others-direct

[DHCPv6]
RouteMetric=5
UseAddress=false
UseDNS=false
UseDelegatedPrefix=true
UseNTP=false

[IPv6AcceptRA]
DHCPv6Client=always
UseDNS=false
```

Ideally, the static routes in the `.network` file should be added to the relevant tables no matter which order pppd and systemd-networkd are started in. 

### Unexpected behaviour you saw

If the ppp interface is created after systemd-networkd has started, the routes won't be added and a restart of systemd-networkd is needed to fix this. Also, I'd like to add that this issue appeared only after I've updated system from 251.1 to 251.10.

### Steps to reproduce the problem

I'm not sure how to easily reproduce this issue, since it requires a pppoe server.

### Additional program output to the terminal or log subsystem illustrating the issue

This is the output of `journalctl -b0 -u systemd-networkd | grep ppp`: https://fars.ee/fWDB

**Comment by YHNdnzj** [comment]
Possibly a duplicate of https://github.com/systemd/systemd/issues/14255?

**Comment by poscat0x04** [comment]
Hmm the symptoms are indeed pretty similar. Maybe this is some sort of regression? Cause I'm pretty sure I had never encountered this issue during version 251.1.

**Comment by poscat0x04** [comment]
Also, what I've noticed is that systemd-networkd does try to do something with the routing tables cause before the ppp interface is created, a `ip r show t <table>` command would return the error
```
Error: ipv4: FIB table does not exist.
Dump terminated
```
But after the ppp interface is created, the same command would show only an empty table.

My guess is that the routes are created and then somehow incorrectly deleted afterwards.

**Comment by poscat0x04** [comment]
This is the output of `ip monitor` after a restart of `pppd.service`:

`systemctl restart pppd.service`
```
inet ppp0 forwarding on rp_filter loose mc_forwarding off proxy_neigh off ignore_routes_with_linkdown off
inet6 ppp0 forwarding on mc_forwarding off proxy_neigh off ignore_routes_with_linkdown off
9: ppp0: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN group default
    link/ppp
9: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN group default
    link/ppp
47.245.48.0/20 dev ppp0 table others-direct proto static scope link
101.101.101.101 dev ppp0 table others-direct proto static scope link
8.209.224.0/19 dev ppp0 table others-direct proto static scope link
47.245.0.0/18 dev ppp0 table others-direct proto static scope link
default dev ppp0 proto static scope link metric 5
47.245.32.0/19 dev ppp0 table others-direct proto static scope link
47.91.16.0/20 dev ppp0 table others-direct proto static scope link
190.115.31.142 dev ppp0 table others-direct proto static scope link
47.74.32.0/19 dev ppp0 table others-direct proto static scope link
47.74.0.0/18 dev ppp0 table others-direct proto static scope link
47.91.0.0/19 dev ppp0 table others-direct proto static scope link
8.209.192.0/18 dev ppp0 table others-direct proto static scope link
8.211.128.0/18 dev ppp0 table others-direct proto static scope link
default dev ppp0 table symmetry proto static scope link
101.6.6.6 dev ppp0 table others-direct proto static scope link
10.1.20.192 dev ens36 lladdr 18:3c:b7:dd:de:c8 PROBE
10.1.20.192 dev ens36 lladdr 18:3c:b7:dd:de:c8 REACHABLE
9: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc fq_codel state UNKNOWN group default
    link/ppp
multicast ff00::/8 dev ppp0 table local proto kernel metric 256 pref medium
fe80::b063:f32e:4856:b56b dev ppp0 proto kernel metric 256 pref medium
9: ppp0    inet6 fe80::b063:f32e:4856:b56b/128 scope link tentative
       valid_lft forever preferred_lft forever
fe80::6f3:52ff:fe94:4bd2 dev ppp0 proto kernel metric 256 pref medium
9: ppp0    inet6 fe80::b063:f32e:4856:b56b peer fe80::6f3:52ff:fe94:4bd2/128 scope link
       valid_lft forever preferred_lft forever
local fe80::b063:f32e:4856:b56b dev ppp0 table local proto kernel metric 0 pref medium
9: ppp0    inet 114.245.107.100/32 scope global ppp0
       valid_lft forever preferred_lft forever
local 114.245.107.100 dev ppp0 table local proto kernel scope host src 114.245.107.100
Deleted 9: ppp0    inet 114.245.107.100/32 scope global ppp0
       valid_lft forever preferred_lft forever
Deleted local 114.245.107.100 dev ppp0 table local proto kernel scope host src 114.245.107.100
Deleted 0.0.0.0 dev ppp0 lladdr 08 NOARP
9: ppp0    inet 114.245.107.100 peer 114.245.104.1/32 scope global ppp0
       valid_lft forever preferred_lft forever
local 114.245.107.100 dev ppp0 table local proto kernel scope host src 114.245.107.100
114.245.104.1 dev ppp0 proto kernel scope link src 114.245.107.100
default via fe80::6f3:52ff:fe94:4bd2 dev ppp0 proto ra metric 5 expires 1800sec mtu 1492 pref medium
2408:8206:4821:afb5::/64 dev ppp0 proto ra metric 5 expires 259200sec pref medium
9: ppp0    inet6 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5/64 scope global tentative dynamic mngtmpaddr noprefixroute
       valid_lft 259199sec preferred_lft 172799sec
9: ppp0    inet6 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 259199sec preferred_lft 172799sec
local 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5 dev ppp0 table local proto kernel metric 0 pref medium
anycast 2408:8206:4821:afb5:: dev ppp0 table local proto kernel metric 0 pref medium
unreachable 2408:8207:482c:ee10::/60 dev lo proto dhcp metric 1024 pref medium
2408:8207:482c:ee10::/64 dev ens36 proto kernel metric 256 expires 259199sec pref medium
3: ens36    inet6 2408:8207:482c:ee10:250:56ff:fe12:ff9a/64 metric 256 scope global tentative dynamic mngtmpaddr
       valid_lft 259199sec preferred_lft 172799sec
3: ens36    inet6 2408:8207:482c:ee10:250:56ff:fe12:ff9a/64 metric 256 scope global dynamic mngtmpaddr
       valid_lft 259198sec preferred_lft 172798sec
local 2408:8207:482c:ee10:250:56ff:fe12:ff9a dev ens36 table local proto kernel metric 0 pref medium
anycast 2408:8207:482c:ee10:: dev ens36 table local proto kernel metric 0 pref medium
10.1.20.86 dev ens36 lladdr e2:ab:02:9c:d8:7b STALE
2408:8207:482b:2110:599b:e695:b45c:1e1d dev ens36 lladdr ae:80:a3:cb:12:72 STALE
10.1.20.118 dev ens36 lladdr ae:80:a3:cb:12:72 STALE
fe80::8b42:e7e9:f149:f4aa dev ens36 lladdr 8e:ed:34:40:24:d1 PROBE
fe80::8b42:e7e9:f149:f4aa dev ens36 lladdr 8e:ed:34:40:24:d1 REACHABLE
Deleted 3: ens36    inet6 2408:8207:482c:ee10:250:56ff:fe12:ff9a/64 metric 256 scope global dynamic mngtmpaddr
       valid_lft 259181sec preferred_lft 172781sec
Deleted anycast 2408:8207:482c:ee10:: dev ens36 table local proto kernel metric 0 pref medium
Deleted local 2408:8207:482c:ee10:250:56ff:fe12:ff9a dev ens36 table local proto kernel metric 0 pref medium
Deleted unreachable 2408:8207:482c:ee10::/60 dev lo proto dhcp metric 1024 pref medium
Deleted 2408:8206:4821:afb5::/64 dev ppp0 proto ra metric 5 expires 259181sec pref medium
Deleted default via fe80::6f3:52ff:fe94:4bd2 dev ppp0 proto ra metric 5 expires 1781sec mtu 1492 pref medium
Deleted 9: ppp0    inet6 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 259181sec preferred_lft 172781sec
Deleted anycast 2408:8206:4821:afb5:: dev ppp0 table local proto kernel metric 0 pref medium
Deleted local 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5 dev ppp0 table local proto kernel metric 0 pref medium
```

`networkctl reconfigure ppp0`
```
47.245.48.0/20 dev ppp0 table others-direct proto static scope link
101.101.101.101 dev ppp0 table others-direct proto static scope link
8.209.224.0/19 dev ppp0 table others-direct proto static scope link
47.245.0.0/18 dev ppp0 table others-direct proto static scope link
default dev ppp0 proto static scope link metric 5
47.245.32.0/19 dev ppp0 table others-direct proto static scope link
47.91.16.0/20 dev ppp0 table others-direct proto static scope link
190.115.31.142 dev ppp0 table others-direct proto static scope link
47.74.32.0/19 dev ppp0 table others-direct proto static scope link
47.74.0.0/18 dev ppp0 table others-direct proto static scope link
47.91.0.0/19 dev ppp0 table others-direct proto static scope link
8.209.192.0/18 dev ppp0 table others-direct proto static scope link
8.211.128.0/18 dev ppp0 table others-direct proto static scope link
default dev ppp0 table symmetry proto static scope link
101.6.6.6 dev ppp0 table others-direct proto static scope link
default via fe80::6f3:52ff:fe94:4bd2 dev ppp0 proto ra metric 5 expires 1800sec mtu 1492 pref medium
2408:8206:4821:afb5::/64 dev ppp0 proto ra metric 5 expires 259200sec pref medium
9: ppp0    inet6 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5/64 scope global tentative dynamic mngtmpaddr noprefixroute
       valid_lft 259199sec preferred_lft 172799sec
9: ppp0    inet6 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 259199sec preferred_lft 172799sec
local 2408:8206:4821:afb5:d96e:ab78:c7c6:5aa5 dev ppp0 table local proto kernel metric 0 pref medium
anycast 2408:8206:4821:afb5:: dev ppp0 table local proto kernel metric 0 pref medium
unreachable 2408:8207:482c:ee10::/60 dev lo proto dhcp metric 1024 pref medium
3: ens36    inet6 2408:8207:482c:ee10:250:56ff:fe12:ff9a/64 metric 256 scope global tentative dynamic mngtmpaddr
       valid_lft 259199sec preferred_lft 172799sec
3: ens36    inet6 2408:8207:482c:ee10:250:56ff:fe12:ff9a/64 metric 256 scope global dynamic mngtmpaddr
       valid_lft 259198sec preferred_lft 172798sec
local 2408:8207:482c:ee10:250:56ff:fe12:ff9a dev ens36 table local proto kernel metric 0 pref medium
anycast 2408:8207:482c:ee10:: dev ens36 table local proto kernel metric 0 pref medium
10.1.20.192 dev ens36 lladdr 18:3c:b7:dd:de:c8 STALE
```

**Comment by poscat0x04** [comment]
Temporary workaround:

Add a script `/etc/ppp/ip-up` with the following content:

```sh
#!/bin/sh
networkctl reconfigure $1
```

Would love to have a proper solution though.
